### ISSUE TITLE
Custom attributes

### ISSUE THREAD

**Issue by schickling** [prisma/prisma#1660]
There are a lot of use cases (e.g. #1604, https://github.com/MichalLytek/type-graphql/issues/476, https://github.com/prisma-labs/nexus-prisma/issues/584, ...) which would benefit from the Prisma schema allowing the definition of "custom attributes" which should be exposed through the Prisma SDK.

**Comment by SrZorro** [comment]
At my company we are currently implementing a custom attribute system for the [typegraphql-prisma](https://github.com/MichalLytek/typegraphql-prisma) generator as it looks like Prisma doesn't have yet a native way to extend it for generator consumers.

We would like to be alienated with the Prisma team in the long run so our extension system maybe could work as a base to what could be brought back to Prisma as a native way to do it.

We have currently [an RFC with our proposal](https://github.com/MichalLytek/typegraphql-prisma/issues/35) that we are already implementing it, but if posible we would like to get feedback from the Prisma team about it.

**Comment by jmarbutt** [comment]
In a lot of applications I have used custom attributes to do a few common things ranging from:

- Defining things like `Description` `Name`  and other basic attributes that may help describe how it is displayed in a UI or for Documentation from the API. 
- Going beyond simple properties we have defined things like `SelectBox` or `DateBox` to go in more detail of how to show these.

An example in .Net we have used is something like this

```c#
[TextBox("Middle Name")] 
public string MiddleName { get; set; }
```

I have used similar patterns in other projects where we allowed this to build up a definition of what a model looks like to the application. This allows models to be consistent across multiple UIs and the model definition contains everything the UI would need to know about how to display the model.
